src/scripts/main.js

```diff
@@ -1 +1,80 @@
 'use strict';
+
+async function wait(ms) {
+  return new Promise((resolve) => setTimeout(resolve, ms));
+}
+
+const firstPromise = new Promise(async (resolve, reject) => {
+  let isTrue = false;
+
+  document.addEventListener('click', () => {
+    isTrue = true;
+    resolve('First promise was resolved');
+  });
+
+  await wait(3000);
+
+  if (!isTrue) {
+    reject(new Error('First promise was rejected'));
+  }
+});
```

Comment:
Bug: The firstPromise will never be rejected. This is because the `await wait(3000);` will always complete before the check for `isTrue`. So, even if the user doesn't click within 3 seconds, the promise will not be rejected. I recommend moving the `await wait(3000);` and the following check for `isTrue` inside the click event listener. This way, the promise will be rejected if the user doesn't click within 3 seconds after the page load.